apps/astarte_appengine_api/lib/astarte_appengine_api_web/router.ex

```diff
@@ -19,106 +19,171 @@ defmodule Astarte.AppEngine.APIWeb.Router do
   use Astarte.AppEngine.APIWeb, :router
   alias Astarte.AppEngine.APIWeb.Plug.LogRealm
 
-  pipeline :api do
+  pipeline :auth_api do
     plug :accepts, ["json"]
     plug LogRealm
     plug Astarte.AppEngine.APIWeb.Plug.AuthorizePath
   end
 
+  pipeline :device_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogDeviceId
+  end
+
+  pipeline :device_group_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogGroupName
+  end
+
+  pipeline :device_alias_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogDeviceAlias
+  end
```

Comment:
Ideally:
- the plug LogDeviceId should only be called for routes that have a `device_id` path param.
- the plug LogGroupName should only be called for routes that have a `group_name` path param.
- the plug LogDeviceAlias should only be called for routes that have a `device_alias` path param.

We could achieve that by using appropiate scopes for the routes, for example:

````elixir
scope "/v1/:realm_name", Astarte.AppEngine.APIWeb do
  pipe_through :realm_api
  
  get "/devices" ...
  
  scope "/devices/:device_id" do
    pipe_through :device_api
    
    get "" ...
    patch "" ...
    
    scope "/interfaces" do
      pipe_through :interface_values_api
      
      get "/" ...
      get "/:interface" ...
      get "/:interface/:path" ...
      put "/:interface/:path" ...
      post "/:interface/:path" ...
      delete "/:interface/:path" ...
    end
  end
  
  scope "/devices-by-alias/:device_alias" do
    pipe_through :device_alias_api
    
    get "" ...
    patch "" ...
    
     scope "/interfaces" do
      pipe_through :interface_values_api
      
      get "/" ...
      get "/:interface" ...
      get "/:interface/:path" ...
      put "/:interface/:path" ...
      post "/:interface/:path" ...
      delete "/:interface/:path" ...
    end
  end
  
  get "/groups" ...
  post "/groups" ...
  
  scope "/groups/:group_name" do
    pipe_through :group_name_api
    
    get "/" ...
    get "/devices" ...
    post "/devices" ...
    
    scope "/devices/:device_id" do
      pipe_through :device_api
    
      delete "/" ...
      
      scope "/interfaces" do
        pipe_through :interface_values_api
      
        get "/" ...
        get "/:interface" ...
        get "/:interface/:path" ...
        put "/:interface/:path" ...
        post "/:interface/:path" ...
        delete "/:interface/:path" ...
      end
    end
  end
end
```